jobrunner/git.py

```diff
@@ -250,7 +253,7 @@ def add_access_token(repo_url):
         return repo_url
     # Ensure we only ever send our token to github.com over https
     parsed = urlparse(repo_url)
-    if parsed.hostname != "github.com" or parsed.scheme != "https":
+    if parsed.hostname != "github-proxy.opensafely.org" or parsed.scheme != "https":
```

Comment:
I know you felt it wasn't worth making this configurable but I think at the very least it needs to be defined as a constant rather than repeated twice. And if we're going to do that then we might as well put it in `config.py` and have it configurable. I also feel like there's an argument that we're supposed to be producing opensource software which can be, at least in theory, used by others and hardcoding opensafely domains into the code feels against that spirit.

Reply:
Hmm, there's a larger question here.

IMO, we are not producing open source software in the traditional sense - at least the job-runner is not - we are building an open source platform and trust around a brand. The open source nature is more about transparency and audit that about 3rd parties consuming the code. I don't see a use case worth our effort to support for anyone else running job-runner outside of the opensafely platform. But that's a probably larger discussion than this review warrants.

Ultimately, I don't think it should to be configurable simply because any other value will a) break and even if it doesn't it b) will not provide the security we have carefully designed in the proxy. The proxy should be the *only* way to access github, as it's the only way we can control and audit access. I see hard coding it as a security feature.

Reply:
I think there's a wider discussion on the open-source question as it relates to issues around the terms of our funding and long term goals, which should be had elsewhere.

In practical terms my ideal would be a env var read in `config.py` with `github-proxy.opensafely.org` as the default (which should protect us against accidental misconfiguration). But I'd settle for a hardcoded value in `config.py`, just so long as there's a single obvious place its defined.

Reply:
Done

Reply:
Thanks for humouring me!

Reply:
Oh man, I don't want to drag this out further but just realised we've still got the domain configured in two places. If you changed the constant to `GIT_PROXY_DOMAIN` without the protocol then you could reuse the it in the `add_access_token` method and do the replace as:
```py
proxied_url = repo_url.replace('https://github.com/', f"https://{config.GIT_PROXY_DOMAIN}/")
```